packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -204,8 +204,13 @@ class DateTimePicker extends React.Component {
       const isDateTimeValid = DateTimeUtils.isValidDateTime(this.dateValue, this.timeValue, this.state.dateFormat, this.props.showSeconds);
       const enteredDateTime = isDateTimeValid ? this.state.dateTime : null;
 
-      this.checkAmbiguousTime(enteredDateTime);
-      this.handleBlur(event, isDateTimeValid);
+      this.checkAmbiguousTime(enteredDateTime, () => {
+        // If the entered time is ambiguous then do not handle blur just yet. It should be handled _after_
+        // the ambiguity is resolved (i.e., after dismissing the Time Clarification dialog).
+        if (!(this.state.isAmbiguousTime && this.state.isTimeClarificationOpen)) {
+          this.handleBlur(event, enteredDateTime);
```

Comment:
We should pass in `this.state.dateTime` as the 2nd parameter to `handleBlur` instead of `enteredDateTime`. The reason is that we want to use the date/time that reflects what the user selected (Daylight Savings or Standard Time) in the Time Clarification dialog. For example, in the Central timezone, if the user selected the `Before (Daylight Saving)` button, the `options.iSO` attribute set in `handleBlur` should have an offset of -500 where as if `After (Standard Time)` was selected, the `options.iSO` attribute should reflect an offset of -600. Passing in `this.state.dateTime` will allow `handleBlur` to use the moment object that reflects what the user selected.

Line 235 should also pass in `this.state.dateTime` as the 2nd parameter to `handleBlur`.

Reply:
That makes sense. Addressed it here: https://github.com/cerner/terra-framework/pull/784/commits/1eaae53d54c7058dd14309b33fd5e0879cea9dd4